src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/InvokerUtil.cs

```diff
@@ -86,31 +87,47 @@ internal static InvokeDelegate GenerateInvokeDelegate(MethodInfo method, out int
             {
                 ParameterInfo[] parameters = method.GetParameters();
                 bool returnsValue = method.ReturnType != typeof(void);
-                int inputCount = parameters.Length;
-                inputParameterCount = inputCount;
-
+                int paramCount = parameters.Length;
+                
+                var inputParamPositions = new List<int>();
                 var outputParamPositions = new List<int>();
-                for (int i = 0; i < inputParameterCount; i++)
+                for (int i = 0; i < parameters.Length; i++)
                 {
                     if (parameters[i].ParameterType.IsByRef)
                     {
-                        outputParamPositions.Add(i);
+                        if (parameters[i].IsOut)
```

Comment:
There's a helper methods to do this for you. ServiceReflector.FlowsIn and ServiceReflector.FlowsOut. These are the methods used for the equivalent functionality on .NET Framework so will make the same decisions.

Reply:
Fixed in the newest commit.